src/components/map/Map.tsx

```diff
@@ -81,8 +116,53 @@ const Map = ({initialLocation, selectionMode, onLocationSelect}: MapProps) => {
           style={{
             flex: 1,
           }}
-          onPress={(feature: Feature) => {
+          onPress={async (feature: Feature) => {
             if (isFeaturePoint(feature)) {
+              const stopPlaceFeature = await findClickedStopPlace(
+                feature,
+                mapViewRef,
+              );
+
+              if (
+                stopPlaceFeature &&
+                showDeparturesBottomSheet &&
+                navigateToDetails &&
+                navigateToQuay
+              ) {
+                openBottomSheet(
+                  (close, focusRef) => (
+                    <DeparturesDialogSheet
+                      bottomSheetSize={bottomSheetSize}
+                      close={close}
+                      stopPlaceFeature={stopPlaceFeature}
+                      navigateToDetails={(
+                        serviceJourneyId: string,
+                        serviceDate: string,
+                        date?: string,
+                        fromQuayId?: string,
+                        isTripCancelled?: boolean,
+                      ) => {
+                        close();
+                        navigateToDetails(
+                          serviceJourneyId,
+                          serviceDate,
+                          date,
+                          fromQuayId,
+                          isTripCancelled,
+                        );
+                      }}
+                      navigateToQuay={(stopPlace, quay) => {
+                        close();
+                        navigateToQuay(stopPlace, quay);
+                      }}
+                      ref={focusRef}
+                    />
+                  ),
+                  closeRef,
+                  false,
+                  bottomSheetStyles.roundEdgesOnTop,
+                );
+              }
```

Comment:
This is too long to have inline in a `onClick` handler. Also we have a hook called `use-selected-feature-change-effect`, which is triggered by the `onMapClick`. Opening the bottom sheet is also an effect which is triggered by a map click, so I think it fits into that hook.

Reply:
@gorandalum do you mean, to move all this inside onMapClick?

Reply:
I have moved to another place using hooks, let me know what do you think ;)